src/TaskLineRenderer.ts

```diff
@@ -169,27 +169,24 @@ export class TaskLineRenderer {
                     span.classList.add(...[componentClass]);
 
                     // Add the component's attribute ('priority-medium', 'due-past-1d' etc.)
-                    const componentDataAttribute = fieldRenderer.dataAttribute(component, task);
-                    for (const key in componentDataAttribute) span.dataset[key] = componentDataAttribute[key];
-                    for (const key in componentDataAttribute) li.dataset[key] = componentDataAttribute[key];
+                    fieldRenderer.addDataAttribute(span, task, component);
+                    fieldRenderer.addDataAttribute(li, task, component);
```

Comment:
I'm not seeing where the `for` loops have moved to...

Reply:
They are not needed actually.

Previously a for loop was used to iterated over `DataAttributes` dictionary. However that dictionary contained only 1 key-value pair. So the for-loop was an implementation detail due to `DataAttributes`. Now it is set directly to the `HTMLElement.dataset`.

The change is in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2442/commits/981670bbe9682ab2315c4ca45ebb627d021b4377

Reply:
You may see in the commit that before running the for-loop over `dataAttribute`, there is just one value set in that dictionary.

```ts
if (this.attributeName !== TaskFieldHTMLData.noAttributeName) {
             const key = this.attributeName;
             const value = this.attributeValueCalculator(component, task);
             dataAttribute[key] = value;
             for (const key in dataAttribute) element.dataset[key] = dataAttribute[key];
         }
```

Reply:
> However that dictionary contained only 1 key-value pair.

And yet the documentation for `TaskFieldRenderer.addDataAttribute()` says plural, suggesting multiple may be added...

> Adds data attributes to an {@link element} for a component.

Reply:
Oh, but the method name is singular `addDataAttribute()` so that is clear. 

I understand now, so am resolving the comment.

I'll stop nit-picking on the JSdocs...

Resolving conversation.

Reply:
It's ok, I will modify the jsdocs, the attributes shall be in single

Reply:
I pushed the jsdoc update